packages/roosterjs-content-model-core/lib/corePlugin/cache/domIndexerImpl.ts

```diff
@@ -232,6 +233,20 @@ export class DomIndexerImpl implements DomIndexer {
             ) {
                 this.reconcileTextSelection(oldSelection.start.node);
             } else {
+                if (
+                    oldSelection.type == 'range' &&
```

Comment:
so many conditions, please extract them to be a separate function, and add comment to explain what it is doing.

Reply:
and please add test case.